### PR TITLE
[BUGFIX beta] added assertion to tell users their arrays need to be wrapped in Ember.A for the #each helper

### PR DESCRIPTION
**Pull request by foxnewsnetwork**
This is in reference to: https://github.com/ember-cli/ember-cli/issues/3734

Where an user (usually an addon developer) would have the each helper complain that the array he passed in is not an array because prototype extensions are not enabled by default anymore in ember-cli scaffolded addons. This assertion message now suggests a solution to his woes.

### CODE REVIEW

File: packages/ember-views/lib/views/each.js [hunk 1]
```diff
@@ -53,17 +53,38 @@ export default CollectionView.extend(_Metamorph, {
...
                      [content.constructor]),
                      !ControllerMixin.detect(content) ||
                        (content && content.isGenerated) ||
                        content instanceof ArrayController);
+
+    hasDetectedArray = EmberArray.detect(content);
+    if (ControllerMixin.detect(content) && content.get('model') !== undefined) {
+      theArrayStr = fmt("'%@' (wrapped in %@)", [content.get('model'), content]);
+      shouldShowError = !Ember.EXTEND_PROTOTYPES &&
+                        Array.isArray(content.get("model")) &&
```
Comment (fivetanley) [inline packages/ember-views/lib/views/each.js]: Array.isArray isn't IE8 safe. We have a helper called `isArray` in ember-metal/lib/utils i think.

**Comment by foxnewsnetwork** [comment]
thank you, will switch over to ember's isArray

**Comment by foxnewsnetwork** [comment]
So I'm doing
```bash
TEST_SUITE=all npm test
```
and getting to the following error
```
Running: phantomjs bower_components/qunit-phantom-runner/runner.js ./dist/tests/index.html?skipPackage=container,ember-testing,ember-debug&dist=min&prod=true

Test failed: global failure
    Failed assertion: Error: Could not find module ember-dev/test-helper/index
file:///home/ia/ember-addons/ember.js/dist/ember-template-compiler.js:55
```
I have no idea what could be causing this; surely, me adding an assertion can't be the culprit.

File: packages/ember-htmlbars/tests/helpers/each_test.js [hunk 2]
```diff
@@ -276,6 +276,35 @@ QUnit.test("View should not use keyword incorrectly - Issue #1315", function() {
   equal(view.$().text(), 'X-1:One 2:Two Y-1:One 2:Two ');
 });
 
+Ember.runInDebug(function() {
```
Comment (rwjblue) [inline packages/ember-htmlbars/tests/helpers/each_test.js]: I do not believe that this is processed properly in tests.  You should use something like the following:

```javascript
// at the top
/* globals EmberDev */

// then in this location:
if (EmberDev && EmberDev.runningProdBuild) {

}
```

**Comment by rwjblue** [comment]
The prod tests will fail unless you do a production build (which is not the default for `npm test`).  You need to do the following:

```
EMBER_ENV=production TEST_SUITE=all npm test
```

File: packages/ember-htmlbars/tests/helpers/each_test.js (hunk 2 above)
Comment (foxnewsnetwork) [inline packages/ember-htmlbars/tests/helpers/each_test.js]: Oh, thank you... but it looks like the tests passed (on Easter Sunday, no less, perhaps I'm on Jesus's good side for once). But should I change out of Ember.runInDebug anyway?

**Comment by foxnewsnetwork** [comment]
Oh, okay, thank you, it works now.

File: packages/ember-htmlbars/tests/helpers/each_test.js (hunk 2 above)
Comment (rwjblue) [inline packages/ember-htmlbars/tests/helpers/each_test.js]: Yes, please.

**Comment by foxnewsnetwork** [comment]
Okay, made the switch to EmberDev and everything now works except the Safari sauce.

**Comment by rwjblue** [comment]
restarted sauce job

**Comment by mixonic** [comment]
@foxnewsnetwork this is looking great, can I beg one more rebase from you?

**Comment by foxnewsnetwork** [comment]
@mixonic ok, on it

**Comment by foxnewsnetwork** [comment]
@mixonic after looking into this, ember's 1.13+ builds no longer require this change because {{each blahs as |blah|}} now runs EnumerableUtil's forEach method, which works on regular arrays as well as EmberArrays. Hooray for {{each}} getting smarter.

**Comment by mixonic** [comment]
@foxnewsnetwork ha! Indeed :-) Thank you for the research and follow up. :cake:

**Comment by foxnewsnetwork** [comment]
@mixonic no problem, thanks to you good people on the core team, ember has become something I use everyday. So the very least I can do is follow up on whatever issues / pr I submit.

**Pull request closed without merging**